ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/explorer/project/ProjectExplorerViewImpl.java

```diff
@@ -555,16 +514,116 @@ public boolean isShowHiddenFiles() {
         return false;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public void onProjectUpdated(ProjectUpdatedEvent event) {
+        ProjectDescriptorNode node = null;
+
+        for (Node seekNode : tree.getNodeStorage().getAll()) {
+            if (seekNode instanceof ProjectDescriptorNode &&
+                ((ProjectDescriptorNode)seekNode).getData().getPath().equals(event.getPath())) {
+                node = (ProjectDescriptorNode)seekNode;
+                break;
+            }
+        }
+
+        if (node == null) {
+            return;
+        }
+
+        final String oldNodeId = tree.getNodeStorage().getKeyProvider().getKey(node);
+
+        Map<String, Node> oldIdToNode = new HashMap<>();
+        for (Node n : tree.getNodeStorage().getAllChildren(node)) {
+            oldIdToNode.put(tree.getNodeStorage().getKeyProvider().getKey(n), n);
+        }
+
+        final ProjectDescriptor updatedDescriptor = event.getUpdatedProjectDescriptor();
+
+        DtoUtil.fill(updatedDescriptor, node.getData());
+
+        if (!tree.getNodeStorage().reIndexNode(oldNodeId, node)) {
+            Log.info(getClass(), "Node wasn't re-indexed");
+        }
+
+        for (Map.Entry<String, Node> entry : oldIdToNode.entrySet()) {
+            if (!tree.getNodeStorage().reIndexNode(entry.getKey(), entry.getValue())) {
+                Log.info(getClass(), "Node wasn't re-indexed");
+            }
+        }
+
+        tree.refresh(node);
+    }
+
```

Comment:
It seems that this method must be located on presenter. The view must be responsible only for view representation.